doc/BUILD.rst

```diff
@@ -0,0 +1,722 @@
+
+Default Renderer (``-Dr_default``)
+''''''''''''''''''''''''''''''''''
+
+* Default: ``gl33``
+* Options: ``gl33``, ``gles30``, ``gles20``, ``null``
+
+Sets the default renderer to use when Taisei launches.
+
+.. code:: sh
+
+   # for GL 3.3 (default)
+   meson configure build/ -Dr_default=gl33
+   # for GL ES 3.0
+   meson configure build/ -Dr_default=gles30
+   # for GL ES 2.0
+   meson configure build/ -Dr_default=gles20
+
+The renderer can be switched in the installed ``taisei`` binary.
+
+ANGLE
+"""""
+
+Shader Transpiler (``-Dshader_transpiler``)
+'''''''''''''''''''''''''''''''''''''''''''
+
+* Default: ``false``
+* Options: ``true``, ``false``
+
+For using ANGLE, the shader transpiler is necessary for converting Taisei's
+shaders to a format usable by that driver.
+
+.. code:: sh
+
+   meson configure build/ -Dshader_transpiler=true
+
+ANGLE Library Paths (``-Dangle_lib*``)
+''''''''''''''''''''''''''''''''''''''
+
+* Default: ``(null)``
+* Options: ``/path/to/libGLESv2.{dll,dylib,so}``/``path/to/libEGL.{dll,dylib,so}``
+
+``-Dangle_libgles`` and ``-Dangle_libegl`` provide the full paths to the ANGLE
+libraries necessary for that engine.
+
+Generally, both need to be supplied at the same time.
+
+.. code:: sh
+
+   # for Linux
+   meson configure build/ -Dangle_libgles=/path/to/libGLESv2.dylib -Dangle_libegl=/path/to/libEGL.dylib
+   # for macOS
+   meson configure build/ -Dangle_libgles=/path/to/libGLESv2.so -Dangle_libegl=/path/to/libEGL.so
+   # for Windows
+   meson configure build/ -Dangle_libgles=/path/to/libGLESv2.dll -Dangle_libegl=/path/to/libEGL.dll
+
+Install ANGLE (``-Dinstall_angle``)
+'''''''''''''''''''''''''''''''''''
+
+* Default: ``false``
+* Options: ``true``, ``false``
+
+Installs the ANGLE libraries supplied above through ``-Dangle_lib*``.
+
+Generally recommended when packaging ANGLE for distribution.
+
+.. code:: sh
+
+   meson configure build/ -Dinstall_angle=true
+
+Building ANGLE (Optional)
+-------------------------
+
+ANGLE is Google's graphics translation layer, intended for for Chromium. Taisei
+packages it with Windows builds to workaround some bugs and performance issues
+with many Windows OpenGL drivers, and it can be optionally packaged as as an
+experimental Metal renderer for macOS.
+
+You'll need to check out
+`ANGLE <https://github.com/google/angle>`__ and build it first. Refer to their
+documentation on how to do that, but generally:
+
+.. code:: sh
+
+   cd angle
+   python ./scripts/bootstrap.py
+   gclient sync
+   gn gen out/x64 --args='is_debug=false dcheck_always_on=false target_cpu="x64"'
+   ninja -C out/x64 libEGL libGLESv2
+
+It will output two files to ``angle/out/x64``:
+
+* ``libEGL.(*)``
+* ``libGLESv2.(*)``
+
+The file extension can be ``.dll`` for Windows, ``.dylib`` for macOS,
+and ``.so`` for Linux.
+
+Using ``-Dinstall_angle`` and ``-Dangle_lib*`` (see above), ``meson`` will copy
+those files over into the package itself when running the packaging steps.
+
```

Comment:
Documenting ANGLE's cursed build process is very out of scope here, and the document is large enough as it is. The explanation of what ANGLE is and why it's used should be moved to the general `ANGLE` section, and then providing [this link](https://github.com/google/angle/blob/master/doc/DevSetup.md) is enough (with a warning that few visitors shall return in one piece and of sound mind).